### PR TITLE
[DOM Reference] Update DOMTokenList and its children

### PR DESCRIPTION
**Pull request by teoli2003**
Part of #9740.

- Update interface and properties/methods to the modern structure
- Removed all mention of `DOMString`
- Removed all use of `var`
- Changed all mentions of `DOMSettableTokenList` that doesn't exist anymore (except one, saying that it doesn't exist anymore :-) )
- Checked/Updated examples.
- Removed all flaws.

There are a few `spec_url` missing in bcd, I'll open a PR to fix them.

### CODE REVIEW

File: files/en-us/web/api/htmlelement/index.md [hunk 1]
```diff
@@ -21,47 +21,47 @@ The **`HTMLElement`** interface represents any [HTML](/en-US/docs/Web/HTML) elem
...
-  - : Is a {{DOMxRef("DOMString")}} representing the access key assigned to the element.
+  - : A {{DOMxRef("DOMString")}} representing the access key assigned to the element.
 - {{DOMxRef("HTMLElement.accessKeyLabel")}} {{ReadOnlyInline}}
   - : Returns a {{DOMxRef("DOMString")}} containing the element's assigned access key.
 - {{DOMxRef("HTMLElement.attributeStyleMap")}} {{ReadOnlyInline}}
-  - : Is a {{DOMxRef("StylePropertyMap")}} representing the declarations of the element's {{htmlattrxref("style")}} attribute.
+  - : A {{DOMxRef("StylePropertyMap")}} representing the declarations of the element's {{htmlattrxref("style")}} attribute.
 - {{DOMxRef("HTMLElement.contentEditable")}}
-  - : Is a {{DOMxRef("DOMString")}}, where a value of `true` means the element is editable and a value of `false` means it isn't.
+  - : A {{DOMxRef("DOMString")}}, where a value of `true` means the element is editable and a value of `false` means it isn't.
```
Comment (wbamberg) [inline files/en-us/web/api/htmlelement/index.md:30]: still a DOMString here?

File: files/en-us/web/api/htmlelement/index.md [hunk 2]
```diff
@@ -21,47 +21,47 @@ The **`HTMLElement`** interface represents any [HTML](/en-US/docs/Web/HTML) elem
...
 - {{DOMxRef("HTMLElement.isContentEditable")}} {{ReadOnlyInline}}
   - : Returns a boolean value indicating whether or not the content of the element can be edited.
 - {{DOMxRef("HTMLElement.contextMenu")}} {{Deprecated_Inline}}
-  - : Is a {{DOMxRef("HTMLMenuElement")}} representing the contextual menu associated with the element. It may be `null`.
+  - : A {{DOMxRef("HTMLMenuElement")}} representing the contextual menu associated with the element. It may be `null`.
 - {{DOMxRef("HTMLElement.dataset")}} {{ReadOnlyInline}}
   - : Returns a {{DOMxRef("DOMStringMap")}} with which script can read and write the element's [custom data attributes](/en-US/docs/Learn/HTML/Howto/Use_data_attributes) (`data-*`) .
 - {{DOMxRef("HTMLElement.dir")}}
-  - : Is a {{DOMxRef("DOMString")}}, reflecting the `dir` global attribute, representing the directionality of the element. Possible values are `"ltr"`, `"rtl"`, and `"auto"`.
+  - : A {{DOMxRef("DOMString")}}, reflecting the `dir` global attribute, representing the directionality of the element. Possible values are `"ltr"`, `"rtl"`, and `"auto"`.
```
Comment (wbamberg) [inline files/en-us/web/api/htmlelement/index.md:38]: ...and here? There are more in this file but I won't comment on every one.

File: files/en-us/web/api/htmliframeelement/index.md [hunk 3]
```diff
@@ -19,45 +19,45 @@ The **`HTMLIFrameElement`** interface provides special properties and methods (b
 _Inherits properties from its parent, {{domxref("HTMLElement")}}_.
 
 - {{domxref("HTMLIFrameElement.align")}} {{deprecated_inline}}
-  - : Is a {{domxref("DOMString")}} that specifies the alignment of the frame with respect to the surrounding context.
+  - : A {{domxref("DOMString")}} that specifies the alignment of the frame with respect to the surrounding context.
```
Comment (wbamberg) [inline files/en-us/web/api/htmliframeelement/index.md:22]: This file too.

File: files/en-us/web/api/htmllinkelement/index.md [hunk 4]
```diff
@@ -20,29 +20,29 @@ The **`HTMLLinkElement`** interface represents reference information for externa
 _Inherits properties from its parent, {{domxref("HTMLElement")}}._
 
 - {{domxref("HTMLLinkElement.as")}}
-  - : Is a {{domxref("DOMString")}} representing the type of content being loaded by the HTML link.
+  - : A {{domxref("DOMString")}} representing the type of content being loaded by the HTML link.
```
Comment (wbamberg) [inline files/en-us/web/api/htmllinkelement/index.md:23]: ...and this one

**Review by wbamberg — CHANGES_REQUESTED** [review]
Thanks @teoli2003 ! I had some nitpicky comments and questions.

File: files/en-us/web/api/domtokenlist/entries/index.md [hunk 5]
```diff
@@ -45,12 +41,12 @@ First, the HTML:
...
-let span = document.querySelector("span");
-let classes = span.classList;
-let iterator = classes.entries();
+const span = document.querySelector("span");
+const classes = span.classList;
+const iterator = classes.entries();
 
 for (let value of iterator) {
-  span.textContent += value + ' ++ ';
+  span.textContent += '(' + value + ') ';
```
[the remark this answers is not in the document]
Comment (teoli2003) [reply files/en-us/web/api/domtokenlist/entries/index.md:49]: Yes, I fixed others (got the idea of a template literal in the middle of the PR). Good catch.

File: files/en-us/web/api/domtokenlist/item/index.md [hunk 6]
```diff
@@ -24,13 +23,17 @@ tokenList.item(index)
 ### Parameters
 
 - `index`
-  - : A 32-bit unsigned integer ({{jsxref("Number")}}) representing the index of the item you want to return.
+  - : A {{jsxref("Number")}} representing the index of the item you want to return. If it isn't an integer, only the integer part is considered.
```
[the remark this answers is not in the document]
Comment (teoli2003) [reply files/en-us/web/api/domtokenlist/item/index.md]: As it is a parameter, it is not (necessarily) a wrapper object, you are right. Fixing.

File: files/en-us/web/api/domtokenlist/index.md [hunk 7]
```diff
@@ -2,44 +2,43 @@
...
+
+A `DOMTokkenList` is indexed beginning with `0` as with JavaScript {{jsxref("Array")}} objects. `DOMTokenList` is always case-sensitive.
 
 ## Properties
 
 - {{domxref("DOMTokenList.length")}} {{ReadOnlyInline}}
   - : Is an `integer` representing the number of objects stored in the object.
 - {{domxref("DOMTokenList.value")}}
-  - : A stringifier property that returns the value of the list as a {{domxref("DOMString")}}.
+  - : A stringifier property that returns the value of the list as a {{jsxref("String")}}.
```
[the remark this answers is not in the document]
Comment (teoli2003) [reply files/en-us/web/api/domtokenlist/index.md]: I checked and it is not a String object. I am a bit surprised, I thought return values were String objects, but it looks like it is not the case, unless maybe rare cases where we have a "live" value. Fixing it.

File: files/en-us/web/api/domtokenlist/item/index.md [hunk 8]
```diff
@@ -24,13 +23,17 @@ tokenList.item(index)
...
 
 ### Return value
 
-A {{domxref("DOMString")}} representing the returned item. It returns
-`null` if the number is greater than or equal to the `length` of
-the list.
+A {{jsxref("String")}} representing the returned item,
```
[the remark this answers is not in the document]
Comment (teoli2003) [reply files/en-us/web/api/domtokenlist/item/index.md]: Fixed.

**Comment by teoli2003** [comment]
I think I have fixed all your comments. Happy to have better understood how `DOMString` is mapped in JS.

**Review by wbamberg — APPROVED** [review]
👍 thanks for the updates!

**Pull request merged**